mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedReceiver.java

```diff
@@ -0,0 +1,93 @@
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import org.openhab.habdroid.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.background.BackgroundUtils.WORKER_TAG_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ENABLED;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ITEM;
+
+public class AlarmChangedReceiver extends BroadcastReceiver {
+    private static final String TAG = AlarmChangedReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d(TAG, "onReceive() with intent " + intent.getAction());
+
+        if (AlarmManager.ACTION_NEXT_ALARM_CLOCK_CHANGED.equals(intent.getAction())
+                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+            if (prefs.getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, false)) {
+                startAlarmChangedWorker(context);
+            }
+        }
+    }
+
+    public static void startAlarmChangedWorker(Context context) {
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+        startAlarmChangedWorker(context, prefs.getString(PREFERENCE_ALARM_CLOCK_ITEM, ""));
+    }
+
+    public static void startAlarmChangedWorker(Context context, String itemName) {
+        Log.d(TAG, "startAlarmChangedWorker()");
+        final Constraints constraints = new Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build();
+        final OneTimeWorkRequest sendAlarmClockWorker =
+                new OneTimeWorkRequest.Builder(AlarmChangedWorker.class)
+                .setConstraints(constraints)
+                .addTag(WORKER_TAG_SEND_ALARM_CLOCK)
+                .build();
+
+        final WorkManager workManager = WorkManager.getInstance();
+        final NotificationManager nm =
+                (NotificationManager)context.getSystemService(NOTIFICATION_SERVICE);
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+
+        Log.d(TAG, "Cancel previous worker");
+        workManager.cancelAllWorkByTag(WORKER_TAG_SEND_ALARM_CLOCK);
+
+        if (TextUtils.isEmpty(itemName.trim())) {
+            Log.d(TAG, "Empty item name");
+            Notification notification = BackgroundUtils.makeBackgroundNotification(context,
+                    R.string.error_sending_alarm_clock_item_empty,
+                    R.drawable.ic_alarm_grey_24dp, true, null);
```

Comment:
Instead of this notification, I'd suggest updating the item name pref summary to reflect the fact that nothing is sent while it's empty.

Reply:
Done, but the notification will still be shown if the alarm changes, but not anymore if the settings are changed.

Reply:
I don't understand ... what exactly do you mean by 'if the alarm changes'? IMHO the worker shouldn't start if the item name is empty, so how would this notification happen?

Reply:
Enabling the feature in the settings or change the item name triggers `startAlarmChangedWorker()`, but won't show any notification if the item name is empty.
Changing the alarm clock time also triggers `startAlarmChangedWorker()` and will show a notification if the item name is empty. IMO a user can overlook the item name preference summary.
In both cases the worker isn't started.

Reply:
I don't think we need to optimize on 'user overlooked item name pref' ... after all, where is the alarm time uploaded to if it's not set? I think we can assume the user knows about the feature's implications when enabling it.

Alternative: make a custom pref with a dialog that includes both switch and editor.

Reply:
I want to keep the settings for now as is and move them to a new page when more intents can be send to the server, e.g. receiving a phone call.